### PR TITLE
VideoCommon: More specific subgroup op bugs

### PR DESCRIPTION
**Pull request by TellowKrinkle**
Previously, subgroup ops only mattered to a tiny performance optimization in bbox handling.

Now, they're used by a very important piece of Dolphin, the ascii art shader.

Split the broken subgroup ops bug into three:
- Completely broken (not currently in use, affects older macOS AMD, but I have no clue which ones as the bug predates any work I've done on the project)
- Broken for how we use it in bbox (affects Intel on macOS, which seems to have trouble only when combined with fragment discard and maybe some other things, as I can't get it to reproduce outside of gameplay)
- is helper invocation check is inverted (newer macOS AMD)

With this, we're able to use the ascii art shader on all GPUs that support subgroup ops in Metal
(Also fixes the Metal code to actually compile the ascii art shader properly)

Note: I added an analytics report for when we notice that our bug tables are incorrect, is that all I need to do there or does the server need to be modified to accept it?

### CODE REVIEW

File: Source/Core/VideoCommon/DriverDetails.cpp [hunk 1]
```diff
@@ -217,4 +221,110 @@ bool HasBug(Bug bug)
...
+    case BUG_SLOW_OPTIMAL_IMAGE_TO_BUFFER_COPY:          return "slow-optimal-image-to-buffer-copy";
+  }
+  return "Unknown";
+}
+
+// clang-format on
+
+void OverrideBug(Bug bug, bool new_value)
+{
+  auto it = m_bugs.find(bug);
```
Comment (iwubcode) [inline Source/Core/VideoCommon/DriverDetails.cpp:308]: Skips the double lookup for the key:

```
const auto [it, added] = m_bugs.try_emplace(bug, BugInfo{m_api, m_os, m_vendor, m_driver, m_family, bug, -1, -1, false});
if (it->second.m_hasbug != new_value)
{
    // report to analytics logic

    it->second.m_hasbug = new_value;
}
```

File: Source/Core/VideoBackends/Metal/MTLUtil.mm [hunk 2]
```diff
@@ -434,22 +507,27 @@ fragment float4 fbfetch_test(float4 in [[color(0), raster_order_group(0)]]) {
   full_source.append(header);
   if (Metal::g_features.subgroup_ops)
     full_source.append(SUBGROUP_HELPER_HEADER);
+  if (DriverDetails::HasBug(DriverDetails::BUG_INVERTED_IS_HELPER))
+    full_source.append("#define gl_HelperInvocation !gl_HelperInvocation "
```
Comment (iwubcode) [inline Source/Core/VideoBackends/Metal/MTLUtil.mm:511]: Multi-lines should be wrapped in `{ }`

(same in vulkan)

File: Source/Core/Core/DolphinAnalytics.h [hunk 3]
```diff
@@ -125,6 +125,9 @@ class DolphinAnalytics
   // to once per game run.
   void ReportGameQuirk(GameQuirk quirk);
 
+  // Get the base builder for building a report
+  const Common::AnalyticsReportBuilder& BaseBuilder() { return m_base_builder; }
```
Comment (iwubcode) [inline Source/Core/Core/DolphinAnalytics.h:129]: `const Common::AnalyticsReportBuilder& BaseBuilder() const { return m_base_builder; }`

**Review by iwubcode — APPROVED** [review]
Code LGTM.  Untested

**Comment by GregoireLD** [comment]
Tried it here using both Metal and Vulkan backends, and various shaders, everything works fine.

**Review by GregoireLD — APPROVED** [review]
LGTM, compiles and runs on MacOS M1. No more shader bugs.

**Review by cesar18pena — APPROVED** [review]

**Pull request merged**